src/TuxMutableRenderMixin.js

```diff
@@ -0,0 +1,131 @@
+  findPath(objectToSearch, [type], mutableTrait);
+  return mutablePath;
+};
+
+// TuxMutableRenderMixin OBJECT: adds mixin to React class that will add componentWillMount and shouldComponentUpdate life cycle events to the component
+module.exports = {
+  // componentWillMount FUNCTION: creates an array of string key paths attached to the component's constructor with a key of __tuxMutableTraits__.
+  // These paths are used in the shouldComponentUpdate life cycle event to compare current and next values of the specified traits to determine
+  // if the component should re-render.
+  componentWillMount: function () {
+    var mutableTraits = this.mutableTraits;
+    var mutableTraitsPaths = [];
+    var type, isArray, isString, mutableTraitsLength, i;
+
+    if (this.constructor.__tuxMutableTraits__) {
+      return;
+    } else {
+      for (type in mutableTraits) {
+        // ensure mutableTraits hasOwnProperty type, otherwise it will walk up the prototype chain
+        if(mutableTraits.hasOwnProperty(type)) {
+          isArray = Array.isArray(mutableTraits[type]);
+          isString = typeof mutableTraits[type] === 'string';
+
+          invariant(isArray || isString, 'mutableTraits needs to be either an array or a string.');
+
+          if (isArray) {
+            mutableTraitsTypeLength = mutableTraits[type].length;
+
+            for (i = 0; i < mutableTraitsTypeLength; i++) {
+              mutableTraitsPaths.push(deepSearch(mutableTraits[type][i], this[type], type));
+            }
+          } else {
+            mutableTraitsPaths.push(deepSearch(mutableTraits[type], this[type], type));
+          }
+
+          // add the trait paths to the constructor object, so only the first component of each class has to perform the deep search
+          this.constructor.__tuxMutableTraits__ = mutableTraitsPaths;
+        }
+      }
+    }
+  },
+
+  // shouldComponentUpdate FUNCTION: add component life cycle event to shouldComponentUpdate that returns a boolean value to determine whether or not the component should update
+  // @param nextProps OBJECT: the updated values for the component's props that will be examined to see if it should re-render
+  // @param nextState OBJECT: the updated values for the component's state that will be examined to see if it should re-render
+  shouldComponentUpdate: function (nextProps, nextState) {
+    // get mutableTraits
+    var mutableTraits = this.constructor.__tuxMutableTraits__;
+
+    // if mutableTraits is present
+    if (mutableTraits) {
+      var mutableTraitsLength = mutableTraits.length;
+      // declare reused variables
+      var traitPathLength, trait, traitPath, currentPathValue, nextPathValue, traitValue;
+
+      // for each trait path see if current is equal to next
+      for (traitPath = 0; traitPath < mutableTraitsLength; traitPath++) {
+        // calculate length of current trait path
+        traitPathLength = mutableTraits[traitPath].length;
```

Comment:
Store result of `mutableTraits[traitPath]` in a variable maybe `mutableTraitPath`.  That way you don't need to keep looking it up here and in the for loop below on line 116.